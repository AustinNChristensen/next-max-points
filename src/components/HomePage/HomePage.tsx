export const HomePage = () => {
    return (
        <div className='text-center'>
            <h1 className='text-4xl tracking-tight font-extrabold text-gray-900 sm:text-5xl md:text-6xl'>
                <span className='block xl:inline'>Data to level up your</span>{' '}
                <span className='block text-blue-600 xl:inline mt-4'>Points Game</span>
            </h1>
            <p className='mt-3 max-w-md mx-auto text-base text-gray-500 sm:text-lg md:mt-5 md:text-xl md:max-w-3xl'>
                Credit cards points is one of the largest arbitrage spaces available today.
            </p>
            <p className='mt-1 max-w-md mx-auto text-base text-gray-500 sm:text-l md:text-xl md:max-w-3xl'>
                Max Points gives you the data you need to make informed decisions on your next card
            </p>
        </div>
    );
};