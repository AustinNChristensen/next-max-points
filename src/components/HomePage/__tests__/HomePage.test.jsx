/**
 * @jest-environment jsdom
 */
import { render, screen } from '@testing-library/react';
import { HomePage } from '../HomePage';

describe('Home Page', () => {
    const setupRTL = () => render(<HomePage />);

    test('matches snapshot', () => {
        const { container } = setupRTL();

        expect(container).toMatchSnapshot();
    });

    test('renders page text', () => {
        setupRTL();

        expect(screen.getByRole('heading', { name: 'Data to level up your Points Game' })).toBeInTheDocument();
        expect(screen.getByText('Credit cards points is one of the largest arbitrage spaces available today.')).toBeInTheDocument();
        expect(screen.getByText('Max Points gives you the data you need to make informed decisions on your next card')).toBeInTheDocument();
    });
});