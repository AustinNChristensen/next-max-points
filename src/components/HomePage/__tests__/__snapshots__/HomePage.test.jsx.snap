// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page matches snapshot 1`] = `
<div>
  <div
    class="text-center"
  >
    <h1
      class="text-4xl tracking-tight font-extrabold text-gray-900 sm:text-5xl md:text-6xl"
    >
      <span
        class="block xl:inline"
      >
        Data to level up your
      </span>
       
      <span
        class="block text-blue-600 xl:inline mt-4"
      >
        Points Game
      </span>
    </h1>
    <p
      class="mt-3 max-w-md mx-auto text-base text-gray-500 sm:text-lg md:mt-5 md:text-xl md:max-w-3xl"
    >
      Credit cards points is one of the largest arbitrage spaces available today.
    </p>
    <p
      class="mt-1 max-w-md mx-auto text-base text-gray-500 sm:text-l md:text-xl md:max-w-3xl"
    >
      Max Points gives you the data you need to make informed decisions on your next card
    </p>
  </div>
</div>
`;
